napari/_vispy/vispy_vectors_layer.py

```diff
@@ -39,3 +39,8 @@ def _on_data_change(self, event=None):
         self.node.update()
         # Call to update order of translation values with new dims:
         self._on_matrix_change()
+
+    def close(self):
+        """Vispy visual is closing."""
+        super().close()
+        self.layer.events.edge_color.disconnect(self._on_data_change)
```

Comment:
This kind of line makes me worry about the automatic name discovery implemented above. I think there's good reasons to not always have a 1-1 correspondence between event names and callback names. So at the very least we should add the ability to override the connection/disconnection names in the utility functions...

Reply:
could go back to this pattern: https://github.com/napari/napari/pull/1984#issuecomment-739436656

Reply:
If we go with this

```python
def disconnect(self, event_group):
    for event in event_group:
        for callback in event.callbacks:
            if hasattr(callback, `__self__`) and callback.__self__ is self:
                disconnect(event, callback)
```
then we don't need to keep track to the list of connections right? we can add connections as we were doing before and they will all just get removed correctly i think / hope?

What do you think of this approach @tlambert03? That might be the simplest (and most magical) change to achieve the desired outcome?

I can give it a try :-)

Reply:
Like it!

Reply:
gone with this pattern